src/librustc_resolve/lib.rs

```diff
@@ -1625,6 +1625,10 @@ pub struct Resolver<'a> {
     current_type_ascription: Vec<Span>,
 
     injected_crate: Option<Module<'a>>,
+
+    /// Is the resolver currently resolving a pattern? Used to silence privacy errors on
+    /// `#[non_exhaustive]` unit and tuple structs when used in `if let` and `match` statements.
+    in_pattern: bool,
```

Comment:
This field is unused now.

Reply:
Oops, thought that got removed in one of the commits I dropped. I'll remove it.

Reply:
Fixed.